xmtp_proto/src/proto_cache.rs

```diff
@@ -0,0 +1,59 @@
+//! Cache for Proto definitions
+
+use prost::Message;
```

Comment:
also moved out of api_d14n to proto crate, changes were made to increase efficiency. the entire lookup table is created on first access.